returnn/datasets/postprocessing.py

```diff
@@ -155,7 +155,7 @@ def __init__(
         for k, t in self._out_tensor_dict_template.data.items():
             if t.vocab:
                 self.labels[k] = t.vocab.labels
-            elif t.sparse_dim:  # sparse_dim but not vocab
+            elif t.sparse_dim and t.sparse_dim.dimension is not None:  # sparse_dim but not vocab
```

Comment:
I don't understand this. You would never have a sparse_dim with `sparse_dim.dimension == None`.

Reply:
Then the dim construction in the PP Dataset from the underlying dataset's data info has a bug. I ran into this as error on a dim by the dataset as part of `_out_tensor_dict_template`. Let me see if I can improve things...

Reply:
Yes that's possible. How does your `sparse_dim` look like? And where does it come from?

Reply:
No wait, this carries raw UTF-8 bytes and has a well-known dimension. This is a bug in PPDataset then.

Reply:
Yeah, it was the orth in OggZipDataset. No bug in PPDataset.

Reply:
So then undo the change here?

Reply:
And also, fix it in `OggZipDataset`?

Reply:
Also, again the question: Where does it come from? We should never even allow to construct a `sparse_dim` with unknown dim here.

Reply:
`OggZipDataset.get_data_shape("orth") -> []` and `OggZipDataset.is_data_sparse("orth") -> false` lead to the only dim being the batch dim created in the line `dims = [Dim(dimension=None, ...)]`:

```python
def _make_tensor_template_from_input(self, data_key: str) -> Tensor:
    dtype = self._dataset.get_data_dtype(data_key)
    if dtype == "string":
        dims = []
    else:
        feature_dims = [
            Dim(dimension=dim, name=f"{data_key}_dim{i + 1}")
            for i, dim in enumerate(self._dataset.get_data_shape(data_key))
        ]
        dims = [Dim(dimension=None, name=f"{data_key}_frame"), *feature_dims]
    sparse_dim = None
    if self._dataset.is_data_sparse(data_key):
        sparse_dim = Dim(dimension=self._dataset.get_data_dim(data_key), name=f"{data_key}_sparse")
        if data_key in self._dataset.labels:
            sparse_dim.vocab = Vocabulary.create_vocab_from_labels(self._dataset.labels[data_key])
    return Tensor(data_key, dims=dims, dtype=dtype, sparse_dim=sparse_dim)
```

So we never constructed a sparse Dim with `dimension=None`. The bug in OggZipDataset is fixed in master already.